### PR TITLE
[Core] separate Communicator exposure

### PR DESCRIPTION
**Pull request by philbucher**
Up until now the `Communicator` is exposed in `add_model_part_to_python`
Now it is moved to its own file, `add_communicator_to_python`

I also took the chance and removed the long deprecated functions that were moved to the `DataCommunicator` a long time ago

### CODE REVIEW

File: kratos/python/add_communicator_to_python.cpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+// Project includes
+#include "includes/define_python.h"
+#include "add_communicator_to_python.h"
+#include "includes/communicator.h"
+
+namespace Kratos {
+namespace Python {
+
+Communicator::MeshType& CommunicatorGetLocalMesh(Communicator& rCommunicator)
```
Comment (loumalouomega) [inline kratos/python/add_communicator_to_python.cpp:25]: Why no lambdas?

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Why no lambdas for a full refactor?

**Comment by philbucher** [comment]
As I [commented earlier](https://github.com/KratosMultiphysics/Kratos/pull/6603#discussion_r398456433) one should not just change code for the sake of changing it !!!!!

Maybe if we can use `pybind::overload_case` then it is worth, but not for an intermediate solution (aka lambdas)

It is ok for NEW code but changing old code has no benefit except the introduction of bugs

**Review by loumalouomega — APPROVED** [review]
Okay

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**